src/fides/api/alembic/migrations/versions/1e750936a141_migrate_to_fideslang_2_0_uses_and_.py

```diff
@@ -0,0 +1,307 @@
+depends_on = None
+
+###############
+## Data Uses ##
+###############
+"""
+The `key` is the old value, the `value` is the new value
+These are ordered specifically so that string replacement works on both parent and child items
+"""
+data_use_upgrades: Dict[str, str] = {
+    "essential.service.operations.support.optimization": "essential.service.operations.improve",
+    "improve.system": "functional.service.improve",
+    "improve": "functional",
+}
+data_use_downgrades: Dict[str, str] = {
+    value: key for key, value in data_use_upgrades.items()
+}
+
+
+def update_privacy_declaration_data_uses(
+    bind: Connection, data_use_map: Dict[str, str]
+) -> None:
+    """Upgrade or downgrade data uses from fideslang 2.0 for privacy declarations"""
+    existing_privacy_declarations: ResultProxy = bind.execute(
+        text("SELECT id, data_use FROM privacydeclaration;")
+    )
+    for row in existing_privacy_declarations:
+        data_use: str = row["data_use"]
+        for key, value in data_use_map.items():
+            if data_use.startswith(key):
+                data_use = data_use.replace(key, value)
+
+        update_data_use_query: TextClause = text(
+            "UPDATE privacydeclaration SET data_use = :updated_use WHERE id= :declaration_id"
+        )
+        bind.execute(
+            update_data_use_query,
+            {"declaration_id": row["id"], "updated_use": data_use},
+        )
+
+
+def update_ctl_policy_data_uses(bind: Connection, data_use_map: Dict[str, str]) -> None:
+    """Upgrade or downgrade data uses from fideslang 2.0 for ctl policies"""
+    existing_ctl_policies: ResultProxy = bind.execute(
+        text("SELECT id, rules FROM ctl_policies;")
+    )
+    for row in existing_ctl_policies:
+        rules: List[Dict] = row["rules"]
+        for i, rule in enumerate(rules or []):
+            data_uses: Dict = rule.get("data_uses", {})
+            for j, val in enumerate(data_uses.get("values", [])):
+                data_use: str = val
+                for key, value in data_use_map.items():
+                    if data_use.startswith(key):
+                        data_use = data_use.replace(key, value)
+                rules[i]["data_uses"]["values"][j] = data_use
+
+        update_data_use_query: TextClause = text(
+            "UPDATE ctl_policies SET rules = :updated_rules WHERE id= :policy_id"
+        )
+        bind.execute(
+            update_data_use_query,
+            {"policy_id": row["id"], "updated_rules": json.dumps(rules)},
+        )
+
+
+#####################
+## Data Categories ##
+#####################
+"""
+The `key` is the old value, the `value` is the new value
+These are ordered specifically so that string replacement works on both parent and child items
+"""
+data_category_upgrades: Dict[str, str] = {
+    "user.financial.account_number": "user.financial.bank_account",
+    "user.credentials": "user.authorization.credentials",
+    "user.observed": "user.behavior",
```

Comment:
Re-surfacing an earlier comment, I'd really like to see this `user.observed` moved to the last entry in this dictionary

Reply:
Done!